multidict/_multidict_c.c

```diff
@@ -0,0 +1,1587 @@
+                               int op)
+{
+    return multidict_tp_richcompare((PyObject*)self->md, other, op);
+}
+
+static void
+multidict_proxy_tp_dealloc(MultiDictProxyObject *self)
+{
+    PyObject_GC_UnTrack(self);
+    PyObject_GC_Del(self);
+}
+
+static int
+multidict_proxy_tp_traverse(MultiDictProxyObject *self, visitproc visit,
+                            void *arg)
+{
+    Py_VISIT(self->md);
+    return 0;
+}
+
+static int
+multidict_proxy_tp_clear(MultiDictProxyObject *self)
+{
+    Py_CLEAR(self->md);
+    return 0;
+}
+
+static PySequenceMethods multidict_proxy_sequence = {
+    0,                                        /* sq_length */
+    0,                                        /* sq_concat */
+    0,                                        /* sq_repeat */
+    0,                                        /* sq_item */
+    0,                                        /* sq_slice */
+    0,                                        /* sq_ass_item */
+    0,                                        /* sq_ass_slice */
+    (objobjproc)multidict_proxy_sq_contains,  /* sq_contains */
+};
+
+static PyMappingMethods multidict_proxy_mapping = {
+    (lenfunc)multidict_proxy_mp_len,          /* mp_length */
+    (binaryfunc)multidict_proxy_mp_subscript, /* mp_subscript */
+    0                                         /* mp_ass_subscript */
+};
+
+static PyMethodDef multidict_proxy_methods[] = {
+    {
+        "getall",
+        (PyCFunction)multidict_proxy_getall,
+        METH_VARARGS | METH_KEYWORDS,
+        multidict_getall_doc
+    },
+    {
+        "getone",
+        (PyCFunction)multidict_proxy_getone,
+        METH_VARARGS | METH_KEYWORDS,
+        multidict_getone_doc
+    },
+    {
+        "get",
+        (PyCFunction)multidict_proxy_getone,
+        METH_VARARGS | METH_KEYWORDS,
+        multidict_get_doc
+    },
+    {
+        "keys",
+        (PyCFunction)multidict_proxy_keys,
+        METH_NOARGS,
+        multidict_keys_doc
+    },
+    {
+        "items",
+        (PyCFunction)multidict_proxy_items,
+        METH_NOARGS,
+        multidict_items_doc
+    },
+    {
+        "values",
+        (PyCFunction)multidict_proxy_values,
+        METH_NOARGS,
+        multidict_values_doc
+    },
+    {
+        "copy",
+        (PyCFunction)multidict_proxy_copy,
+        METH_NOARGS,
+        multidict_copy_doc
+    },
+    {
+        "__reduce__",
+        (PyCFunction)multidict_proxy_reduce,
+        METH_NOARGS,
+        NULL
+    },
+    {
+        NULL,
+        NULL
+    }   /* sentinel */
+};
+
+static PyTypeObject multidict_proxy_type = {
+    PyVarObject_HEAD_INIT(NULL, 0)
+    "multidict._multidict.MultiDictProxy",           /* tp_name */
+    sizeof(MultiDictProxyObject),                    /* tp_basicsize */
+    0,                                               /* tp_itemsize */
+    (destructor)multidict_proxy_tp_dealloc,          /* tp_dealloc */
+    0,                                               /* tp_print */
```

Comment:
Mb we should change this comment and other the same comments to /* tp_vectorcall_offset */ according to https://www.python.org/dev/peps/pep-0590/#id12?

Reply:
Good idea